lib/puppet/util/windows/adsi.rb

```diff
@@ -201,21 +233,50 @@ def remove_from_groups(*group_names)
     end
     alias remove_from_group remove_from_groups
 
+
+    def add_group_sids(*sids)
+      group_names = []
+      sids.each do |sid|
+        group_names << Puppet::Util::Windows::SID.sid_to_name(sid)
+      end
+
+      add_to_groups(*group_names)
+    end
+
+    def remove_group_sids(*sids)
+      group_names = []
+      sids.each do |sid|
+        group_names << Puppet::Util::Windows::SID.sid_to_name(sid)
+      end
+
+      remove_from_groups(*group_names)
+    end
+
+    def group_sids
+      self.class.get_sids(native_user.Groups)
+    end
+
     def set_groups(desired_groups, minimum = true)
       return if desired_groups.nil? or desired_groups.empty?
+      # this needs fixed up once PUP-3653 is merged up
```

Comment:
That is this commit https://github.com/puppetlabs/puppet/commit/6b8d187cd151f01ddcaf43f26354ee6755f6c4c6 (with at the current time is only in 3.x)